### ISSUE TITLE
QForm.submit and QForm.validate do not respect async rules

### ISSUE THREAD

**Issue by holtalanm** [quasarframework/quasar#5432]
**Describe the bug**
A form with async rules.  Calling submit on that form when one or more of those rules would result in an error still submits the form.

**Codepen/jsFiddle/Codesandbox (required)**
https://codepen.io/holtalanm/pen/eYYeygM

**To Reproduce**
1. Enter a value into the Name field
2. Wait for it to auto-submit
3. Notice age field failed validation, but submit notification was still shown, indicating that the form was submitted.

**Expected behavior**
Age field validation should fail, but submit notification should NOT be shown, as the form should not be submitted when there is failed validation.

Contrast this behavior with synchronous validation:
https://codepen.io/holtalanm/pen/ZEEaoao

**Platform (please complete the following information):**
OS: Windows 10
Browsers: Firefox 70

**Comment by rstoenescu** [comment]
Hi,

Thanks for reporting!
Fix will be available in "quasar" v1.3.0.

**Comment by holtalanm** [comment]
Thank you!

**Comment by OfekA-IAI** [comment]
@rstoenescu This is happening to me with Quasar 2.4.4 - any fix for this?

**Comment by BenJackGill** [comment]
I'm having the same problem.

When my input has an async rule validation the form will submit even though it should throw an error.

**Comment by OfekA-IAI** [comment]
@BenJackGill I wasn't able to find the cause of the issue, but in my case I was able to "solve" it like this:

I had a parent and a child component.
The parent had the form components and the `.validate()` call.
The child had a loading indicator which was triggered by a prop from the parent.

After removing that prop that was changed before and after validation ,`.validate()`  seems to work correctly

I still think there's some other issue here but couldn't spend the time to find out.

Hope this helps

**Comment by holtalanm** [comment]
I ended up 'solving' this issue by rolling my own validation, in the end.

**Comment by BenJackGill** [comment]
In my case the QForm was submitting before the QInput async validation had finished.

So I used the async validation inside the `@submit` of the form instead. Now it works.

This bug still needs looking at though.

@rstoenescu Can we reopen this issue?

**Comment by rstoenescu** [comment]
This works correctly. Just checked it.
Please fill a codepen so I can help.

**Comment by BenJackGill** [comment]
**Describe the bug:** A form with async rules will submit even if it has validation errors. This only happens on first page load and when pressing the enter key. All subsequent form submits work as expected.

**Codepen/jsFiddle/Codesandbox (required)**
I could not get it working on codepen, so I spun up a brand new Quasar CLI app and tested on that instead. Full details below.

**To Reproduce**
Steps to reproduce the behavior:

1) Create a new Quasar app using Quasar CLI command `quasar create <app-name-here>`. Settings chosen: Sass with SCSS Syntax, EsLint, TypeScript, Composition API, Prettier, NPM.
2) Create a new component `FormTest.vue` inside `src/components` using the below code:

```
<template>
  <q-form @submit="handleSubmit">
    <q-input
      label="Input"
      v-model="modelValue"
      :rules="validationRules"
      lazy-rules
    />
    <q-btn type="submit" label="Submit" />
  </q-form>
</template>

<script setup lang="ts">
import { ref } from 'vue';

const modelValue = ref('');

//Simulating long database call
const stall = async (stallTime = 3000) => {
  await new Promise((resolve) => setTimeout(resolve, stallTime));
};
const fakeDatabaseCallReturnFalse = async () => {
  await stall(); // stalls for the default 3 seconds
  return false;
};

//The validation rules
const validationRules = [
  async () =>
    (await fakeDatabaseCallReturnFalse()) ||
    'Fake databse call returned false.',
];

//The form submit
const handleSubmit = () => {
  console.log('The form was submitted!');
};
</script>

```
3) Replace `src/pages/index.vue` with the code below (to show the form):

```
<template>
  <q-page class="row items-center justify-evenly">
    <form-test />
  </q-page>
</template>

<script lang="ts">
import { defineComponent } from 'vue';
import FormTest from 'src/components/FormTest.vue';

export default defineComponent({
  name: 'PageIndex',
  components: { FormTest },
  setup() {
    //
  },
});
</script>

```
4) Startup quasar using the `quasar dev` command
5) Visit the page in browser with the console open. Type some text into the input box and press ENTER (do not click the submit button).

After the 3 second wait the input will show a validation error, but the console will show that the form was successfully submitted.

**Expected behaviour:**  The form should not submit because there was an input validation error.

Note: The bug only happens on first page load and pressing the enter key. 

If you try to submit the form again (a second time after page load) then it works as expected. Also if you click the submit button instead of pressing the enter key then it works as expected.

**Comment by rstoenescu** [comment]
Thank you for supplying the detailed explanation!
Fix will be available in Quasar v2.5.1 and v1.18.1
